### ISSUE TITLE
[date-picker] component is not accessible via screenreader

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#5310]
## Environment
Tested on macOS latest in Safari for VoiceOver and generic keyboard testing and Window 10 latest in Firefox for JAWS.

Testing https://carbon-components-react.netlify.com (unreleased latest)

### VoiceOver 🍎 
- The calendar dropdown is not announced when the input is focused
- If the user focuses the calendar VO announces it as a "group inside of web content" with no further instruction
- The user is not informed dates can be selected
- When the user selects a date the selection is not announced; the focused date is simply read again
- When a date is selected and the user is selecting a subsequent range none of this behavior is announced (same as above)

### JAWS 🦈 
_note: JAWS was so broken that testing most of the aforementioned interactions wasn't opossible_
- Calendar dropdown was not announced
- Calendar can be accessed with the down arrow, but once the calendar date is focused dates can't be changed using arrow keys (seems to be clashing with JAWS reading commands as groups are being focused and read. @snidersd can you confirm this behavior?)

**Comment by snidersd** [comment]
Testing this with VO, macOS and Chrome with the following issues:
1. After a date is selected from the calendar VO announces that it is invalid data. This happens in all date picker calendars.
<img width="1440" alt="range-calendar" src="https://user-images.githubusercontent.com/21676914/74190706-67a4c880-4c18-11ea-98f5-d97875f689a9.png">
2. Date in the calendar should be selectable with Space and Enter key and it should close the dialog after the date is selected. Space does not work and the user must hit the Enter key twice to close the dialog. In addition, the second time the Enter key is hit it also populates the second date in the date range.

3. Range calendar with min/max - displays the January calendar when it opens and does not display the min/max range until the second date picker is selected. 
<img width="1047" alt="Screen Shot 2020-02-10 at 3 07 36 PM" src="https://user-images.githubusercontent.com/21676914/74191562-763faf80-4c19-11ea-81fc-96d29561692c.png">
4. Fully controlled - 
Arrow down to dates and VO starts announcing every date in the calendar.
<img width="1172" alt="Screen Shot 2020-02-10 at 3 10 22 PM" src="https://user-images.githubusercontent.com/21676914/74191637-996a5f00-4c19-11ea-8b30-8d9b0c3af488.png">

Here is a link to the WAI-ARIA date picker pattern for keyboard accessible keyboard controls: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html

**Comment by asudoh** [comment]
Some info wrt the keyboard navigation:

* The current range version lets user select both start/end first time the date picker is opened. Looks that this is more of a UX decision.
* Not being able to re-open the date picker by keyboard after the first selection (by enter key) seems simply a bug that needs to be fixed.
* Ctrl-arrow keys can be used to increment/decrement year/month by keyboard (Mac Mission Control may take over that keyboard shortcut. You can disable it if it's the case)

**Comment by obro** [comment]
Team, do you have a target ETA for fix implementation?

**Comment by obro** [comment]
@asudoh Curious if any comment? We use this UI control in Your Learning.

**Comment by dakahn** [comment]
Tested with VO in Chrome and Safari

After an update to latest Flatpickr I couldn't replicate the severity:1 bugs. There was still some awkwardness with Chrome: 
- hitting enter twice to close date selection dialog
- on second date selection focus is returned to first date input

That said it tested well in Safari -- our preferred browser pairing with VO. This big broad issue is a little unhelpful, so I'll open a ticket to track the Chrome bugs specifically and close this one out. Thanks all! 👍🏽
